packages/core/src/ui/UIImage.ts

```diff
@@ -0,0 +1,283 @@
+      this._tiledAdaptiveThreshold = value;
+      if (this.drawMode === SpriteDrawMode.Tiled) {
+        this._dirtyUpdateFlag |= ImageUpdateFlags.VertexData;
+      }
+    }
+  }
+
+  /**
+   * The Sprite to render.
+   */
+  get sprite(): Sprite {
+    return this._sprite;
+  }
+
+  set sprite(value: Sprite | null) {
+    const lastSprite = this._sprite;
+    if (lastSprite !== value) {
+      if (lastSprite) {
+        this._addResourceReferCount(lastSprite, -1);
+        lastSprite._updateFlagManager.removeListener(this._onSpriteChange);
+      }
+      this._dirtyUpdateFlag |= ImageUpdateFlags.All;
+      if (value) {
+        this._addResourceReferCount(value, 1);
+        value._updateFlagManager.addListener(this._onSpriteChange);
+        this.shaderData.setTexture(UIRenderer._textureProperty, value.texture);
+      } else {
+        this.shaderData.setTexture(UIRenderer._textureProperty, null);
+      }
+      this._sprite = value;
+    }
+  }
+
+  /**
+   * Rendering color for the Sprite graphic.
+   */
+  get color(): Color {
+    return this._color;
+  }
+
+  set color(value: Color) {
+    if (this._color !== value) {
+      this._color.copyFrom(value);
+      this._dirtyUpdateFlag |= ImageUpdateFlags.Color;
+    }
+  }
+
+  /**
+   * @internal
+   */
+  constructor(entity: Entity) {
+    super(entity);
+
+    this.drawMode = SpriteDrawMode.Simple;
+    this._dirtyUpdateFlag |= ImageUpdateFlags.Color;
+    this.setMaterial(this._engine._uiDefaultMaterial);
+    this._onSpriteChange = this._onSpriteChange.bind(this);
+  }
+
+  /**
+   * @internal
+   */
+  protected override _render(context: RenderContext): void {
+    const { _sprite: sprite } = this;
+    const transform = this._transform as UITransform;
+    const { x: width, y: height } = transform.size;
+    if (!sprite?.texture || !width || !height) {
+      return;
+    }
+
+    let material = this.getMaterial();
+    if (!material) {
+      return;
+    }
+    // @todo: This question needs to be raised rather than hidden.
+    if (material.destroyed) {
+      material = this._engine._uiDefaultMaterial;
+    }
+
+    const { _dirtyUpdateFlag: dirtyUpdateFlag } = this;
+    // Update position
+    if (dirtyUpdateFlag & ImageUpdateFlags.Position) {
+      this._assembler.updatePositions(this, width, height, transform.pivot);
+    }
+
+    // Update uv
+    if (dirtyUpdateFlag & ImageUpdateFlags.UV) {
+      this._assembler.updateUVs(this);
+    }
+
+    // Update color
+    if (dirtyUpdateFlag & ImageUpdateFlags.Color) {
+      this._assembler.updateColor(this, this._alpha);
+    } else if (dirtyUpdateFlag & UIRendererUpdateFlags.Alpha) {
+      this._assembler.updateAlpha(this, this._alpha);
+    }
+
+    this._dirtyUpdateFlag = ImageUpdateFlags.None;
+    // Init sub render element.
+    const { engine } = context.camera;
+    const canvas = this._canvas;
+    const subRenderElement = engine._subRenderElementPool.get();
+    const subChunk = this._subChunk;
```

Comment:
We can further optimize by destructuring primitive from subChunk.chunk to reduce the nested property access. Here's an improved version:

const { _canvas: canvas, _subChunk: subChunk, sprite } = this;
const { primitive } = subChunk.chunk;
const subRenderElement = engine._subRenderElementPool.get();
const texture = sprite.texture;

subRenderElement.set(this, material, primitive, subChunk.subMesh, texture, subChunk);